tests/testFilteringCatalogs.py

```diff
@@ -0,0 +1,591 @@
+                return np.where(base % 2 == 0, base, None)
+
+            @cached
+            def get_ip3t(self):
+                base = self.column_by_name('ip3')
+                return np.where(base % 3 == 0, base, None)
+
+        cat_name = os.path.join(self.scratch_dir, "inst_post_facto_filter_cat.txt")
+        if os.path.exists(cat_name):
+            os.unlink(cat_name)
+
+        cat = FilteredCat3(self.db, cannot_be_null=['ip2t', 'ip3t'])
+        cat.write_catalog(cat_name)
+
+        with open(cat_name, 'r') as input_file:
+            input_lines = input_file.readlines()
+
+        self.assertEqual(len(input_lines), 3)  # two data lines and a header
+        for i_line, line in enumerate(input_lines):
+            if i_line is 0:
+                continue
+            else:
+                ii = (i_line - 1)*6
+                ip1 = ii + 1
+                ip2 = ii + 2
+                ip3 = ii + 3
+                self.assertEqual((ii+2) % 2, 0)
+                self.assertEqual((ii+3) % 3, 0)
+                self.assertEqual(line,
+                                 '%d, %d, %d, %d\n' % (ii, ip1, ip2, ip3))
+
+        if os.path.exists(cat_name):
+            os.unlink(cat_name)
+
+    def test_compound_column(self):
+        """
+        Test filtering on a catalog with a compound column that is calculated
+        after the filter column (example code has shown this to be a difficult case)
+        """
+
+        class FilteredCat4(InstanceCatalog):
+            column_outputs = ['id', 'a', 'b', 'c', 'filter_col']
+            cannot_be_null = ['filter_col']
+
+            @compound('a', 'b', 'c')
+            def get_alphabet(self):
+                ii = self.column_by_name('ip3')
+                return np.array([ii*ii, ii*ii*ii, ii*ii*ii*ii])
+
+            @cached
+            def get_filter_col(self):
+                base = self.column_by_name('a')
+                return np.where(base % 3 == 0, base/2.0, None)
+
+        cat_name = os.path.join(self.scratch_dir, "inst_compound_column_filter_cat.txt")
+        if os.path.exists(cat_name):
+            os.unlink(cat_name)
+
+        cat = FilteredCat4(self.db)
+        cat.write_catalog(cat_name)
+
+        with open(cat_name, 'r') as input_file:
+            input_lines = input_file.readlines()
+
+        # verify that the catalog contains expected data
+        self.assertEqual(len(input_lines), 5)  # 4 data lines and a header
+        for i_line, line in enumerate(input_lines):
+            if i_line is 0:
+                continue
+            else:
+                ii = (i_line - 1)*3
+                ip3 = ii + 3
+                self.assertEqual((ip3*ip3) % 3, 0)
+                self.assertEqual(line, '%d, %d, %d, %d, %.1f\n'
+                                        % (ii, ip3*ip3, ip3*ip3*ip3, ip3*ip3*ip3*ip3, 0.5*(ip3*ip3)))
+
+        if os.path.exists(cat_name):
+            os.unlink(cat_name)
+
+    def test_filter_on_compound_column(self):
+        """
```

Comment:
What about a test on a compound catalog where (using the 'a','b','c' you have here) the column 'c' has null values, but 'a' or 'b' cannot be null and testing that nothing gets filtered? I admit I'm not sure what all gets used as a compound column in practice, but I see this as a possible theoretical case that hasn't been tested.